### DISCUSSION TITLE
How to use github search query to get the number of usage of my language

### DISCUSSION THREAD

**Discussion by pewulfman** [github-linguist/linguist#5736] [Q&A]
The contribution guideline clearly state :
```
We try only to add languages once they have some usage on GitHub. In most cases, we prefer that each new file extension is in use in at least 200 unique :user/:repo repositories before supporting them in Linguist.
```
```
    Open a pull request, linking to GitHub search results showing in-the-wild usage. 
```

But the example you provide and the experimentation I made seems to indicate that you can research file in all GitHub that use an extension, but not a list of repositories that contains at least one file using that extension.

So, how can I do the correct search results ? 
I guess I can make a script that will scroll the result of the extension search to get a list of unique `:user/:repo` but shouldn't your search engine be able to do that ?

best

**Comment by Nixinova** [comment]
You're right, there is no way to get a repository count other than downloading all the search results. That's the main issue with contributing to this project.

**Reply by Nixinova** [reply]
There are some tools like https://github.com/alhadis/harvester that are made to do this task but they don't always work and GitHub rate limits you quite a bit.

**Reply by pewulfman** [reply]
Is there a plan to add this functionality to the GitHub search engine? Can someone external (like me) propose a merge request?

**Reply by lildude** [reply]
I have no idea (as I'm not a member of the team that maintains Search) but Search is undergoing a major rewrite with the new "Technology Preview" Search available at https://cs.github.com/. Whilst you can't propose a PR, you can provide feedback via the various links.
